NctApiClientLibrary/Classes/NctSessionManager.psm1

```diff
@@ -34,23 +34,40 @@ class NctSessionManager {
                 Write-Verbose "Using existing valid session"
                 return $this.Session
             }
-
+            
             # Get credentials
             $credentials = $this.GetCredentials()
             if (-not $credentials) {
                 throw "Failed to get credentials"
             }
 
+            # Check if 2FA is required
+            $oneTimePassword = $this.Check2FA($credentials)
+
+            # Build request 
+            $body = 
+@"
+{
+    "UserName": $($credentials.username),
+    "Password": $($credentials.Password),
+    "RememberMe": false,
+    "Meta": {
+        "OneTimePassword": $oneTimePassword
+    }
+}
+"@
+            
             # Create new session
             $uri = "$($this.HubUrl)/auth/credentials"
```

Comment:
I'm not seeing the same requirement we have in the agents use of the API, namely, mandatory use off HTTP**S** for the hub url.  Especially as this is a security sensitive PR this seems like a good time to fix this.

If necessary I'd be OK with this requirement only applying to remote hosts (i.e. HTTP and HTTPS acceptable on localhost (etc), HTTPS required otherwise).